app/usecase/keys/consumer_test.go

```diff
@@ -1 +1,115 @@
 package keys
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/byliuyang/kgs/app/entity"
+	"github.com/byliuyang/kgs/app/usecase/repo/repotest"
+
+	"github.com/byliuyang/app/mdtest"
+)
+
+func TestCachedConsumer(t *testing.T) {
+	keys := []entity.Key{
+		"aaaa", "aaab", "aaac", "aaad", "aaae", "aaaf", "aaag", "aaah",
+		"baaa", "baab", "baac", "baad", "baae", "baaf", "baag", "baah",
+		"caaa", "caab", "caac", "caad", "caae", "caaf", "caag", "caah",
+	}
+
+	availableKeysRepo := repotest.NewAvailableKeyFake()
+	allocatedKeysRepo := repotest.NewAllocatedKeyFake()
+
+	for _, key := range keys {
+		err := availableKeysRepo.Create(key)
+		mdtest.Equal(t, nil, err)
+	}
+
+	mockConsumer := NewConsumerPersist(
+		&availableKeysRepo,
+		&allocatedKeysRepo,
+	)
+
+	testCases := []struct {
+		name     string
+		expected fakeConsumerResult
+		count    uint
+	}{
+		{
+			name: "Should load keys and return three items from the buffer #1",
```

Comment:
```suggestion
			name: "should load keys and return three items from the buffer #1",
```

Nit: Let's start with lowercase for consistency.